utils/buffer/examples/buffer_pool_exhaustion.rs

```diff
@@ -0,0 +1,65 @@
+// # Handling Buffer Pool Exhaustion and Heap Allocation
+//
+// This example demonstrates how a buffer pool is filled. The back slots of the buffer pool are
+// exhausted first, followed by the front of the buffer pool. Once both the back and front are
+// exhausted, data is allocated on the heap at a performance decrease.
+//
+// 1. Fills up the back slots of the buffer pool until they’re exhausted.
+// 2. Releases one slot to allow the buffer pool to switch to front mode.
+// 3. Fully fills the front slots of the buffer pool.
+// 4. Switches to alloc mode for direct heap allocation when both the buffer pool's back and front
+//    slots are at capacity.
+//
+// # Run
+//
+// ```
+// cargo run --example buffer_pool_exhaustion
+// ```
+
+use buffer_sv2::{Buffer, BufferPool};
+use std::collections::VecDeque;
+
+fn main() {
+    // 8 byte capacity
+    let mut buffer_pool = BufferPool::new(8);
+    let mut slices = VecDeque::new();
+
+    // Write data to fill back slots
+    for _ in 0..4 {
+        let data_bytes = b"a"; // 1 byte
+        let writable = buffer_pool.get_writable(data_bytes.len()); // Mutable slice to internal
+                                                                   // buffer
+        writable.copy_from_slice(data_bytes);
+        let data_slice = buffer_pool.get_data_owned(); // Take ownership of allocated segment
+        slices.push_back(data_slice);
```

Comment:
Or maybe you're adding the last 4 bytes here?

Reply:
@rrybarczyk did you miss this one?

Reply:
see https://github.com/stratum-mining/stratum/pull/1232/files#r1865236367